src/unidiff.c

```diff
@@ -202,7 +202,7 @@ int diff_loadAvailable (void)
       }
 
       diff = &array_grow(&diff_available);
-      diff->filename = strdup( diff_files[i] );
+      diff->filename = diff_files[i];
```

Comment:
Maybe just to be cautious, this line should be followed with `diff_files[i] = NULL;` so it isn't possible to try to use the pointer after it's been "moved".

Not that I see a significant danger of that here, but it may be a good practice.

Reply:
> I wonder if instead of `if (ptr != NULL) free(ptr);`, it would be better to just use `free(ptr);`. The C standard guarantees that it's valid to free null pointers, and it's the preferred style in at least some style guides. But I guess the more verbose version may be the style Naev's code is already using.

This is validating to see; I considered raising that point on a previous round. (While compatibility is very important, if Naev can use variadic macros, I think it can rely on this part of ISO/IEC 9899 too.)
For the moment, though, avoiding free(NULL) is a pervasive pattern in this code base.

> Maybe just to be cautious, this line should be followed with diff_files[i] = NULL; so it isn't possible to try to use the pointer after it's been "moved".

I like that pattern. (In this file, we free diff_files a few lines below, so I'm kind of indifferent. Would be happy to edit that in tomorrow though.)